dotnet/src/SemanticKernel.Abstractions/AI/Embeddings/Embedding.cs

```diff
@@ -46,38 +46,47 @@ public Embedding(IEnumerable<TEmbedding> vector)
 
         if (!IsSupported)
         {
-            throw new NotSupportedException($"Embeddings do not support type '{typeof(TEmbedding).Name}'. "
-                                            + $"Supported types include: [ {string.Join(", ", Embedding.SupportedTypes.Select(t => t.Name).ToList())} ]");
+            ThrowNotSupportedEmbedding();
         }
 
         // Create a local, protected copy
         this._vector = vector.ToArray();
     }
 
+    private static void ThrowNotSupportedEmbedding() =>
+        throw new NotSupportedException($"Embeddings do not support type '{typeof(TEmbedding).Name}'. Supported types include: [ Single, Double ]");
+
     /// <summary>
-    /// Gets the vector as a <see cref="ReadOnlyCollection{T}"/>
+    /// Gets the vector as an <see cref="IEnumerable{TEmbedding}"/>
     /// </summary>
     [JsonPropertyName("vector")]
-    public IEnumerable<TEmbedding> Vector => this._vector.AsEnumerable();
```

Comment:
I just realized the intent here with the AsEnumerable may have been to return am enumerable that wraps the array so that the array itself isn't exposed. But AsEnumerable os just a cast, so it was a nop here. If the goal is to return a wrapper, Select can be used.

Reply:
I recall this going through a few iterations in the interest of having it be visible to the json serializer without a full copy. Whatever the case -- looks good.